src/utilities/effects.ts

```diff
@@ -4652,15 +4658,40 @@ const effects = {
       if (query !== null) {
         const viewIdAsNumber = getSearchParameterNumber(SearchParameters.VIEW_ID, query);
 
+        // Derive view from url or model default
+        let view;
         if (viewIdAsNumber !== null) {
           const data = await reqHasura<View>(gql.GET_VIEW, { id: viewIdAsNumber }, user);
-          const { view } = data;
+          const { view: fetchedView } = data;
+          view = fetchedView;
+        } else if (defaultView !== null && defaultView !== undefined) {
+          view = defaultView;
+        }
 
-          if (view !== null) {
+        if (view) {
+          // Return view if not asked to migrate the view
+          if (!migrate) {
             return view;
           }
-        } else if (defaultView !== null && defaultView !== undefined) {
-          return defaultView;
+
+          // Otherwise perform any needed migrations
+          const { migratedView, error, anyMigrationsApplied } = await applyViewMigrations(view);
+          if (migratedView && anyMigrationsApplied) {
+            await effects.updateView(
+              migratedView.id,
+              { definition: migratedView.definition },
+              'View Automatically Migrated',
+              user,
+            );
+          }
+
+          // If migration failed catch the error and return default view
+          if (!migratedView) {
+            catchError('Unable to migrate view', error as Error);
+            showFailureToast(`Unable to migrate view: ${view.name}`);
```

Comment:
Should the error also include the "automatic" wording?

Reply:
Yeah I'm not opposed to that.